app/services/foreman_ansible/variables_importer.rb

```diff
@@ -48,22 +48,21 @@ def import_new_role(role_name, new_roles)
     def initialize_variables(variables, role)
       variables.map do |variable_name, variable_default|
         variable = AnsibleVariable.find_or_initialize_by(
-          :key => variable_name
+          :key => variable_name,
+          :ansible_role_id => role.id
         )
-        if variable.new_record?
-          variable.assign_attributes(:default_value => variable_default,
-                                     :key_type => infer_key_type(variable_default),
-                                     :imported => true)
-        end
-        variable.ansible_role = role
+        variable.assign_attributes(:default_value => variable_default,
+                                   :key_type => infer_key_type(variable_default))
+        variable.imported = true if variable.new_record?
         variable.valid? ? variable : nil
       end
     end
 
     def detect_changes(imported)
       changes = {}.with_indifferent_access
-      persisted, changes[:new] = imported.partition { |role| role.id.present? }
-      changes[:update], _old = persisted.partition(&:changed?)
+      persisted, changes[:new] = imported.partition { |var| var.id.present? }
+      changed, _old = persisted.partition(&:changed?)
+      _overriden, changes[:update] = changed.partition(&:override?)
```

Comment:
I am not opposed to it, but why not to update the default value for overriden variables? Should this fix [#26485](https://projects.theforeman.org/issues/26485)?

Reply:
Well, because if the value has `override` flag, the `default_value` is set by user :thinking:
So I left those updates for a followup. Such update would effectively mean `reimport the variable and delete my changes to it` I just thought it's too dangerous :)

Reply:
So let's do it in next PR.